### ISSUE TITLE
deis client can't setup ssl connection via cloudflare

### ISSUE THREAD

**Issue by szymonpk** [deis/deis#3938]
For testing purposed we're using Cloudflare with their SSL service. However deis client can't setup secure connection to controller it's throwing:
```
$ deis apps
Couldn't connect to the Deis Controller:
[Errno bad handshake] [('SSL routines', 'SSL23_GET_SERVER_HELLO', 'sslv3 alert handshake failure')]
Make sure that the Controller URI is correct and the server is running.
```

I can do `curl https://<my controller>/v1/` without any issues. I've thought it may be due to crappy old openssl on OSX, I've installed fresh one from brew, but no-go. I've tried to install PyOpenSSL with proper LDFLAGS `env ARCHFLAGS="-arch x86_64" LDFLAGS="-L/usr/local/opt/openssl/lib" CFLAGS="-I/usr/local/opt/openssl/include" pip install PyOpenSSL`, this doesn't work either.

Any hints or ideas?

**Comment by bacongobbler** [comment]
Cloudflare has [disabled SSLv3](https://mo.nitor.me/what-cdn-provider-is-still-using-sslv3/) due to the [POODLE exploit](https://isc.sans.edu/forums/diary/SSLv3+POODLE+Vulnerability+Official+Release/18827/). I guess the client's still trying to make an SSLv3 connection, which you'd have to switch over to TLSv1, TLSv1.1 or TLSv1.2.

**Comment by szymonpk** [comment]
I'm not sure how to force tls with deis client, libssl-dev isn't osx package, I've tried to upgrade openssl (via brew package manager) and force PyOpenSSL to use it. I've also tried to build deis client from source, and if I use `./dists/deis apps` I'm getting error like [this](https://gist.github.com/szymonpk/f78466c01b19a4bb52fa).

However if I use `./deis.py apps` (uncompiled version), everything works like a charm.
